clouddriver/clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/docker/HelmOciArtifactConfiguration.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2025 Harness, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.artifacts.docker;
+
+import com.netflix.spinnaker.credentials.CredentialsTypeProperties;
+import com.netflix.spinnaker.kork.client.ServiceClientProvider;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.OkHttpClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ConditionalOnExpression(
+    "${kubernetes.enabled:false} || ${cloudrun.enabled:false} || ${dockerRegistry.enabled:false}")
+@EnableConfigurationProperties({
+  HelmOciDockerArtifactProviderProperties.class,
+  HelmOciArtifactProviderProperties.class
+})
+@RequiredArgsConstructor
+@Slf4j
+class HelmOciArtifactConfiguration {
+
+  private final HelmOciDockerArtifactProviderProperties helmOciDockerArtifactProviderProperties;
+
+  private final HelmOciArtifactProviderProperties helmOciArtifactProviderProperties;
```

Comment:
Same question here about adding args to the bean methods that need these objects.  Then we don't need `RequiredArgsConstructor` either.

Reply:
Good catch, modified